src/Microsoft.Extensions.OptionsModel/OptionsWatcher.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Extensions.OptionsModel
+{
+    public class OptionsWatcher<TOptions> : IOptionsWatcher<TOptions> where TOptions : class, new()
+    {
+        private TOptions _options;
+        private IEnumerable<IConfigureOptions<TOptions>> _setups;
+        private IOptionsChangeTracker<TOptions> _tracker;
+
+        // REVIEW: IEnumerable monitors?
+        public OptionsWatcher(IEnumerable<IConfigureOptions<TOptions>> setups, IOptionsChangeTracker<TOptions> tracker)
```

Comment:
So we'd basically wire up an OnChange for each tracker that fires the Action, and keep a list of the IDisposables and return a new one that wraps disposing all of the ones we created?